### PR TITLE
Don't lookup github projects not in layout.

### PR DESCRIPTION
**Pull request by jamielennox**
Gerrit does a funny thing by fetching change information you receive -
even if you don't have that project in your layout. This is supposed to
update the change cache for any cross-project dependencies on foreign
projects. The project object is None that gets passed to getChange and
gerrit only uses event information to fetch data.

We can't do this with github yet, but what we get is a

    'NoneType' object has no attribute 'name'

message when you try to fetch this empty project. Probably we should be
fetching based on event.project_name but really we can't support it
anyway so just short circuit and return a NullChange.

Change-Id: Ib03c428e7fe82bd62530f0a8fc834dae6648fd58
Signed-off-by: Jamie Lennox <jamielennox@gmail.com>

### CODE REVIEW

File: zuul/source/github.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 import logging
 import time
 
-from zuul.model import PullRequest, Ref
+from zuul.model import NullChange, PullRequest, Ref
```
Comment (omgjlk) [inline zuul/source/github.py:18]: This will be fun to port, as there is no NullChange in v3 any more :/

**Review by omgjlk — APPROVED** [review]

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491339087.87/tox_py27 : FAILURE in 2m 44s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491339261.94/tox_py27 : FAILURE in 2m 38s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491341324.91/tox_py27 : FAILURE in 2m 40s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491341719.52/tox_py27 : FAILURE in 50s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491341939.29/tox_py27 : FAILURE in 48s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491342164.09/tox_py27 : FAILURE in 49s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491342389.03/tox_py27 : RETRY_LIMIT

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491344413.24/tox_py27 : FAILURE in 2m 42s

**Comment by gandelman-a** [comment]
gate-recheck

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491345565.18/bonnyci-run-gate : FAILURE in 13s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491345590.83/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491345613.64/bonnyci-run-gate : FAILURE in 11s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491345633.07/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491345651.21/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491345670.19/bonnyci-run-gate : FAILURE in 11s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491345689.61/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491345707.5/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491345727.35/bonnyci-run-gate : FAILURE in 10s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491345745.86/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491345764.11/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491345788.77/bonnyci-run-gate : FAILURE in 10s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491345814.77/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491345839.29/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491345856.37/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491345873.43/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491345893.6/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491345912.67/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491345930.33/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491345947.95/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491345969.18/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491345986.91/bonnyci-run-gate : FAILURE in 10s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491346005.11/bonnyci-run-gate : FAILURE in 10s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491346023.86/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491346041.7/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491346059.81/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491346081.1/bonnyci-run-gate : FAILURE in 13s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491346102.09/bonnyci-run-gate : FAILURE in 10s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491346126.43/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491346145.74/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491346163.43/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491346180.64/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491346198.28/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491346216.36/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491346233.8/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491346251.73/bonnyci-run-gate : FAILURE in 10s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491346270.74/bonnyci-run-gate : FAILURE in 10s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491346288.75/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491346306.37/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491346324.78/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491346342.79/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491346360.75/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491346378.28/bonnyci-run-gate : FAILURE in 9s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491346395.94/bonnyci-run-gate : FAILURE in 10s

**Comment by anne-bonny** [comment]
Build failed.

- http://logs.bonnyci.org/gate_github/BonnyCI/zuul/36/1491346413.81/bonnyci-run-gate : FAILURE in 9s

**Review by gandelman-a — CHANGES_REQUESTED** [review]
-1'ing this to get it out of the gate

File: zuul/source/github.py (hunk 1 above)
Comment (jamielennox) [reply zuul/source/github.py:18]: The NullChange gets dropped in the return, so you could return anything here and it would basically get ignored.

**Review by gandelman-a — APPROVED** [review]

**Pull request merged**